tests/src/Context/FieldContext.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\nexteuropa\Context\FieldContext.
+ */
+
+namespace Drupal\nexteuropa\Context;
+
+use Behat\Behat\Context\Context;
+
+/**
+ * Context with field management.
+ */
+class FieldContext implements Context {
+
+  /**
+   * The list of field instances before a scenario starts.
+   *
+   * @var array
+   */
+  protected $defaultFields = array();
+
+  /**
+   * The list of field groups before a scenario starts.
+   *
+   * @var array
+   */
+  protected $defaultFieldGroups = array();
+
+  /**
+   * Remember the list of fields existing by default in the site.
+   *
+   * @BeforeScenario
+   */
+  public function rememberCurrentFields() {
+    $this->defaultFields = field_info_field_map();
+  }
+
+  /**
+   * Removes any fields created by a scenario.
+   *
+   * @AfterScenario
+   */
+  public function resetFields() {
+    if (empty($this->defaultFields)) {
+      return;
+    }
+
+    $current_fields = field_info_field_map();
+    $purge_field_schema = FALSE;
+    foreach ($current_fields as $field_name => $field_info) {
+      if (!isset($this->defaultFields[$field_name])) {
+        $field_info = field_info_field($field_name);
+        if (!is_null($field_info)
+          && !$field_info['locked']
+          && !$field_info['deleted']) {
```

Comment:
I put a question on the doc because I had to implement that after meeting problems and the documentation does not say "field_info_field only return a non-deleted fields", it says "$field_name can only refer to a non-deleted, active field." and "NULL if the field was not found." that is not precisely the same.